internal/middlewares/authelia_context.go

```diff
@@ -563,25 +563,41 @@ func (ctx *AutheliaCtx) AcceptsMIME(mime string) (acceptsMime bool) {
 }
 
 // SpecialRedirect performs a redirect similar to fasthttp.RequestCtx except it allows statusCode 401 and includes body
-// content in the form of a link to the location.
+// content in the form of a link to the location if the request method was not head.
 func (ctx *AutheliaCtx) SpecialRedirect(uri string, statusCode int) {
+	var u []byte
+
+	u, statusCode = ctx.setSpecialRedirect(uri, statusCode)
+
+	ctx.SetContentTypeTextHTML()
+	ctx.SetBodyString(fmt.Sprintf("<a href=\"%s\">%d %s</a>", utils.StringHTMLEscape(string(u)), statusCode, fasthttp.StatusMessage(statusCode)))
+}
+
+// SpecialRedirectNoBody performs a redirect similar to fasthttp.RequestCtx except it allows statusCode 401 and includes
+// no body.
+func (ctx *AutheliaCtx) SpecialRedirectNoBody(uri string, statusCode int) {
+	_, _ = ctx.setSpecialRedirect(uri, statusCode)
```

Comment:
Would it be better to just factorize this into a single function based on the method being `HEAD` that way we can simplify all of the instances where it's called?

Reply:
I did this specifically for the authz endpoints as the method to the endpoint doesn't necessarily match the original method. I did have in mind doing what you suggested but in the instance of  something like HAProxy's lua module it will always be HEAD for example.